java/sage/MiniClientSageRenderer.java

```diff
@@ -1328,6 +1328,33 @@ private boolean checkForAudioOutputChangeUpdates()
     return true;
   }
 
+  private boolean checkForMenuHintUpdates()
+  {
+    String hintText;
+    synchronized (menuHint) {
+      hintText=menuHint.format();
+      menuHint.clear();
+    }
+    if (hintText!=null)
```

Comment:
You should still call hasHint() here...then get the hintText in the sync block after that. With this current implementation...hintText will never be null (since format doesn't look at the state) and you'll get the HINT property on every single command sequence sent over. :)